### PR TITLE
feature: articles with `redirect: /Page` will redirect to specified page

### LINKED ISSUE #73: feature request: redirect
sometimes I'd like to preserve links (e.g. to `/Posts/About`, but move the actual page to a different path `/About`).  It would be nice - since we're now supporting more than Markdown files, to support a `redirect: /About` (seems to be yaml syntax which we agree on atm), which will lead to a `301 Moved Permanently` HTTP response.

does this sound like a good idea?

### PR DESCRIPTION
**Pull request by gjaldon**
Closes #73 

Thanks a lot for the help, @hannesm!! My first ever OCaml open-source contribution!!! 🙇 🎆

### CODE REVIEW

File: canopy_store.ml [hunk 1]
```diff
@@ -102,8 +111,7 @@ module Store (CTX: Irmin_mirage.CONTEXT) (INFL: Git.Inflate.S) = struct
         match C.of_string ~base_uuid ~uri ~content ~created ~updated with
         | C.Ok article -> KeyMap.add key (`Article article) cache
         | C.Error error ->
-          Log.warn (fun f -> f "Error while parsing %s: %s" uri error) ;
-          cache
+          check_if_redirect content cache key (fun f -> f "Error while parsing %s: %s" uri error)
```
Comment (hannesm) [inline canopy_store.ml]: I do like this PR -- but can we have `check_redirect` be `string -> string option` (as in, pass only `content` and return either `None` or `Some path`, and do the `add` or `warn` still in `fill_cache`?

**Comment by gjaldon** [comment]
@hannesm I just applied the change you suggested. :)

**Comment by hannesm** [comment]
great, thanks! :)

**Comment by gjaldon** [comment]
my pleasure 🙇

**Comment by hannesm** [comment]
thanks again for your contribution, I now use this on https://hannes.nqsb.io (now having an /About and /Posts/About redirecting there).  I did minor modifications to your implementation, https://github.com/Engil/Canopy/commit/eb41a0c1aa88b47fd54b30754ccc0dce49260b02, namely adding the `~sep` label to have OCaml not complain, and `String.trim` the value of the redirect.

and on top of that https://github.com/Engil/Canopy/commit/b93dad3591a1d1edfbd998dc651b7a44ea8a416a to separate the concerns a bit stricter..

**Comment by gjaldon** [comment]
That's great you're already using the feature! :) Will try migrating my blog over to Canopy on my next weekend just to try out mirage in 'prod'.

Thanks for sharing the modifications you did. Should've noticed the warning and the other commit was particularly useful as reference for good style.

**Pull request merged**